lib/run_part.c

```diff
@@ -13,12 +16,16 @@
 #include "run_part.h"
 #include "shadowlog_internal.h"
 
-int run_part (char *script_path, const char *name, const char *action)
+#ifdef HAVE_EXECVEAT
+int run_part (int script_fd, const char *script_name, const char *name, const char *action)
+#else
+int run_part (const char *script_path, const char *script_name, const char *name, const char *action)
+#endif			/* HAVE_EXECVEAT */
```

Comment:
How about using `/proc/self/fd/n`?  That wouldn't need execveat(2), so would be more portable.

Reply:
Heh, I was implementing my own fexecve() that did that, and after doing it, I searched out of curiosity if anyone had implemented a fexecve(), and found that there exists a libc function with that name, and it's in POSIX.1-2008.  :)

<https://www.man7.org/linux/man-pages/man3/fexecve.3.html>